cron.js

```diff
@@ -26,101 +29,205 @@ class CronManager {
+                this.db.delete(tenant, jobId).then(() => {
+                    resolve({jobId: jobId, spec: value.spec});
+                }).catch(error => {
+                    logger.error(`Failed to unset cron job ${jobId} (${error}).`);
+                    reject(new InternalError(`Internal error while unsetting cron job ${jobId}.`));
+                });
 
-                resolve(jobId);
+                logger.info(`Succeeded to unset cron job ${jobId}.`);
             }
-            catch(ex) {
-                console.warn(`Failed to schedule job ${Json.stringify(jobSpec)} (${ex})`);
-                reject(new InternalError('Internal Error'));
+            else {
+                logger.debug(`Not found job ${jobId} for tenant ${tenant}`);
+                reject(new JobNotFound(`Not found job ${jobId} for tenant ${tenant}`));
             }
         });
     }
 
+    init() {
+        logger.info('Initializing cron service ...');
+        
+        // dojot messenger
+        return this.dojotMessenger.init().then(()=>{
+            logger.info('Communication with dojot messenger service (kafka) was established.');
+            return this.brokerHandler.init();
+        // handler for broker jobs
+        }).then(() => {
+            logger.info('Handler for broker jobs was initialized.');
+            return this.db.init();
+        //database
+        }).then(()=> {
+            logger.info('Communication with database (mongoDB) was established.');
+
+            // tenancy channel
+            this.dojotMessenger.createChannel(
+                config.kafkaMessenger.dojot.subjects.tenancy, "r", true /*global*/);
+            logger.info('Read-only channel for tenancy events was created.');
+
+            // tenancy channel: new-tenant event
+            this.dojotMessenger.on(config.kafkaMessenger.dojot.subjects.tenancy, 
+                'new-tenant', (_tenant, newtenant) => {
+                    this._setTenant(newtenant);
+            });
+            
+            let tenantSetPromises = [];
+            for(let tenant of this.dojotMessenger.tenants) {
+                tenantSetPromises.push(this._setTenant(tenant));
+            }
+            return Promise.all(tenantSetPromises);      
+        }).catch(error => {
+            // something unexpected happended!
+            logger.error(`Couldn't initialize the cron manager (${error}).`);
+            return Promise.reject(new InternalError('Internal error while starting cron manager.'));
+        });
+    }
+
+    createJob(tenant, jobSpec) {
+        return new Promise((resolve, reject) => {
+            // job id
+            let jobId = uuidv4();
+
+            // db -job
+            let dbEntry = {
+                jobId: jobId,
+                spec: jobSpec
+            };
+            this.db.create(tenant, dbEntry).then(() => {
+                this._setCronJob(tenant, jobId, jobSpec).then(() => {
+                    resolve(jobId);
+                });
```

Comment:
I think is missing a `catch` clause here to treat when the _setCronJob fails.

Reply:
You're right!